source/sonata_simulation.rst

```diff
@@ -263,6 +263,22 @@ Series of current pulse injections.
    frequency                      float      Mandatory    The frequency of pulse trains. Given in Hz.
    represents_physical_electrode  boolean    Optional     Default is False. If True, the signal will be implemented using a NEURON IClamp mechanism. The IClamp produce an electrode current which is not included in the calculation of extracellular signals, so this option should be used to represent a physical electrode. If the noise signal represents synaptic input, `represents_physical_electrode` should be set to False, in which case the signal will be implemented using a  MembraneCurrentSource mechanism, which is identical to IClamp, but produce a membrane current, which is included in the calculation of the extracellular signal. 
    ============================== ========== ============ ==========================================
+   
+sinusoidal (current_clamp)
+~~~~~~~~~~~~~~~~~~~~~
+
+Series of current pulse injections.
+
+.. table::
+
+   ============================== ========== ============ ==========================================
+   Property                       Type       Requirement  Description
+   ============================== ========== ============ ==========================================
+   amp_start                      float      Mandatory    The peak amplitude of the sinusoid. Given in nA.
+   frequency                      float      Mandatory    The frequency of the waveform. Given in Hz.
+   dt                             float      Optional     Timestep of generated signal in ms. Default is 0.25 ms.
```

Comment:
Is being able to specify a `dt` other than the simulation one important?
Is 0.25ms chosen because it is the simulation `dt`?
If so, should it default to whatever the simulation `dt` is?

Reply:
0.25 ms was chosen because it is the default for the noise stimuli. I am not sure if being able to specify dt actually is important for the sinusoidal stim

Reply:
Actually I asked him to add it since in neurodamus I saw:
```
# generate sinusoidal current source
                cs = CurrentSource.sin(
                    self.amp, self.duration, self.freq, step=self.dt, delay=self.delay,
                    physical_electrode=self.represents_physical_electrode
                )
...
def parse_check_all_parameters(self, stim_info: dict):
        self.dt = float(stim_info.get("Dt", 0.025))  # stimulus timestep [ms]
```

But checking the BlueConfig documentation: https://sonata-extension.readthedocs.io/en/latest/blueconfig.html
I see dt is only needed for:
```
Dt | For Noise, ShotNoise, RelativeShotNoise, AbsoluteShotNoise, OrnsteinUhlenbeck and RelativeOrnsteinUhlenbeck stimuli, the timestep of the current or conductance to inject.
```

So maybe is not needed in this case? Thoughts?

Reply:
Looks like the dt is passed directly to:
https://nrn.readthedocs.io/en/8.1.0/python/programming/math/vector.html#Vector.sin

So having it makes sense.
What the suitable `dt` is, is up to you; I was just wondering how it interacts w/ the simulation dt.